src/main/java/com/iexec/sms/secret/compute/TeeTaskComputeSecretService.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2021 IEXEC BLOCKCHAIN TECH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.iexec.sms.secret.compute;
+
+import com.iexec.sms.encryption.EncryptionService;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.data.domain.Example;
+import org.springframework.data.domain.ExampleMatcher;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Slf4j
+@Service
+public class TeeTaskComputeSecretService {
+    private final TeeTaskComputeSecretRepository teeTaskComputeSecretRepository;
+    private final EncryptionService encryptionService;
+
+    protected TeeTaskComputeSecretService(
+            TeeTaskComputeSecretRepository teeTaskComputeSecretRepository,
+            EncryptionService encryptionService) {
+        this.teeTaskComputeSecretRepository = teeTaskComputeSecretRepository;
+        this.encryptionService = encryptionService;
+    }
+
+    /**
+     * Retrieve a secret.
+     * Decrypt if required.
```

Comment:
I thought the secret would be always decrypted when fetched from the db!

Reply:
Good catch!